File: Python/ValidPerfectSquare.py
def validPerfectSquare(num):
    l,r = 1,num

    while l<=r:
        mid = (l+r)//2
        if mid*mid>num:
            r=mid-1
        elif mid*mid<num:
            l=mid+1
        else:
            return True

    return False

File: Javascript/ValidPerfectSquare.js
var validPerfectSquare = function(num){
    let l = 1
    let r = num
    
    while(l<=r){
        mid = Math.floor((l+r)/2)
        if(mid*mid>num){
            r=mid-1
        }
        else if(mid*mid<num){
            l=mid+1
        }
        else{
            return true
        }
    }

    return false
}